src/kitovu/sync/settings.py

```diff
@@ -0,0 +1,105 @@
+"""A collection of all settings wrappers and factories for the sync."""
+
+import pathlib
+import typing
+
+import yaml
+import attr
+
+from kitovu import utils
+
+
+SimpleDict = typing.Dict[str, typing.Any]
+
+
+@attr.s
+class PluginSettings:
+    """A class representing the settings of a signle plugin"""
+
+    plugin_type: str = attr.ib()
+    connection: SimpleDict = attr.ib()
+    syncs: typing.List[SimpleDict] = attr.ib(default=[])
+
+
+@attr.s
+class Settings:
+    """A class representing the settings of a all plugins"""
+
+    root_dir: pathlib.PurePath = attr.ib()
+    global_ignore: typing.List[str] = attr.ib()
+    plugins: typing.Dict[str, PluginSettings] = attr.ib()
+
+
+class YAMLSettingsFactory:
+    """A factory to create settings objects from files or strings.
+
+    It cannot be done in the Settings class itself because mypy does
+    not know the Settings type inside of the class.
+    """
+
+    @classmethod
+    def from_file(cls, path: pathlib.PurePath) -> Settings:
```

Comment:
in order to improve readability of the code, rename "cls" to something more easily readable - I can't derive its meaning or what it's supposed to stand for.

Reply:
@bytinbit Thank you for the typos, I've changed them.

`cls` is basically the same as `self` just for class methods. It is common to use `cls` instead of `class`, because `class` is a reserved keyword and stuff like `current_class` or similiar names are too verbose to be used frequently.